proper_parenthetics.py

```diff
@@ -1,34 +1,29 @@
 # _*_ utf-8 _*_
 """Interview Challenge: Proper Parenthetics."""
+from collections import Counter
 
 
 def parens(string):
     """Check a string to make sure correct number of parenthesis are there."""
-    count = 0
-    for idx in string:
-        if idx is u"(":
-            count += 1
-        elif idx is u")":
-            count -= 1
-
-    if count > 0:
-        print(1)
+    paren_dict = Counter(string)
+    if paren_dict[")"] < paren_dict["("]:
```

Comment:
Here's a test case that should fail, but will not under this approach:

```
"))))(((("
```

That is a "broken" string, in that the first close parens appears before the first open parens, but because your implementation is only checking if the counts of open and closed parens are equal it will pass this as good.  

Can you think of how you might change the code so that this case is detected as it should be?  Here are a few other cases you need to be able to detect:

```
"())("
"(()))("
```